### ISSUE TITLE
Log file is not created if it doesn't exist

### ISSUE THREAD

**Issue by SpcCw** [friendica/friendica#13537]
I am not sure if it is a bug report or just a question so let's assume the best :)

I am pretty sure that before the Friendica created the log file if it did not exist and logging was enabled. I even have some rotation mechanism which moves log files every hour and new one used to be created.

Now I see that if the log file is missing it doesn't get created even though there is write permission for www-data. If I create it then it starts filling up normally until moved/deleted.

Is it some intended change, bug or problem with my system?

**Comment by annando** [comment]
Looks more like a problem on your system. I recommend using logrotate. My file looks like this:
```
/path/to/friendica/logs/friendica.log {
        rotate 2
        daily
        missingok
        compress
        delaycompress
        create 660 user group
        postrotate
                su user -c "/path/to/friendica/htdocs/bin/daemon.php start"
        endscript
        prerotate
                su user -c "/path/to/friendica/htdocs/bin/daemon.php stop"
        endscript
}
```

**Comment by SpcCw** [comment]
@annando I see you have `create` in your config. Does it mean Friendica isn't supposed to create log file if it doesn't exist?

Because it definitely did so in the past :) It isn't a big thing of course but got me wondering.

**Comment by annando** [comment]
I use `create` to ensure that the file permissions is correct.

**Comment by haheute** [comment]
@SpcCw (I am on the rc version..) I have just tried and moved `friendica.log` to `friendica.log.backup`. After some time and clicking, a new `friendica.log` appeared. Have you tried to "save settings" again in `/admin/logs` ?

with this logrotate file:
```
/var/log/friendica/friendica.log {
	rotate 5
	compress
	delaycompress
}
```

**Comment by SpcCw** [comment]
@haheute - now it is getting strange because [I asked this on Friendica Admins group](https://forum.friendi.ca/display/d0446be5-1965-2bc6-1fe8-083952205821) and two other admins confirmed the log not getting created.

Do you have the latest pull?

**Comment by haheute** [comment]
my latest entry in `git log` is this:

commit a5fb1e53a991e9e23424405c8dc5a0b5e9a030d8 (HEAD -> 2023.09-rc, origin/2023.09-rc)
Merge: e99fcfddb a1fb0f848
Author: Michael Vogel <icarus@dabo.de>
Date:   Sun Oct 15 08:03:50 2023 +0200

    Merge pull request #13544 from MrPetovan/bug/13542-frio-circle-search
    
    [frio] Add search in contact address to filter circle member list

**Comment by haheute** [comment]
I have also clicked "save settings" in `/admin/logs`. And the logfile that appeared had size `0`when it appeared..

**Comment by SpcCw** [comment]
@haheute now I am puzzled... and yes, I am sure changes are applied because if I change log file name in log settings then View Logs section reflects it. Also if I change log file name and create the file then it gets filled.

What PHP version do you have? It is one thing I changed recently (everything else seems to work though).

**Comment by haheute** [comment]
I have `8.1.23` (Alpine Linux)

**Comment by haheute** [comment]
I have tried once again (`mv friendica.log friendica.log.backup`) and the `friendica.log` was newly created after I visited `admin/logs/view`. Without clicking "save settings" anywhere.

**Comment by AlfredSK** [comment]
I didn't expect that Friendica is creating the file. So I always created the file manually and then let logrotate do the rotation and re-creating.

**Comment by SpcCw** [comment]
A little update: I played with it for a while (temporary changing permissions to 777, rebooting system, starting/stopping Friendica daemon, changing log file location) and while I still can't get anything consistent and 99% of the time Friendica doesn't create the log a couple of times it did.

It is weird. Not a big thing certainly but weird.

For now I just changed log rotation to recreate the file and it all works smoothly. I am closing this not to distract everyone from more important things and if I manage to reproduce it or have some ideas I'll create proper issue/bug report. Thanks for the help, everyone!
